File: .vendor/dislin/julia/areaf.jl
include("Dislin.jl")

ix  = [0, 300, 300,   0]
iy  = [0,   0, 400, 400]

ixp  = Array{Int32}(undef, 4)
iyp  = Array{Int32}(undef, 4)

Dislin.scrmod("revers")
Dislin.metafl("cons")
Dislin.disini()
Dislin.setvlt("small")
Dislin.pagera()
Dislin.complx()

Dislin.height(50)
ctit = "Shading patterns (AREAF)"
n = Dislin.nlmess(ctit)
Dislin.messag(ctit, div(2970 - n, 2), 200)

nx0 = 335
ny0 = 350

for i = 0:2
  ny = ny0 + i * 600

  for j = 0:5
    nx = nx0 + j * 400
    ii = i * 6 + j
    x = i * 6.0 + j

    Dislin.shdpat(ii)
    iclr = i * 6 + j + 1
    iclr = iclr % 8
    if (iclr == 0)
      iclr = 8
    end 
    Dislin.setclr(iclr)
    for k = 1:4
      ixp[k] = ix[k] + nx
      iyp[k] = iy[k] + ny
    end

    Dislin.areaf(ixp, iyp, 4)
    nl = Dislin.nlnumb(x, -1)
    nx = nx + div(300 - nl, 2)
    Dislin.color("foreground")
    Dislin.number(x, -1, nx, ny + 460) 
  end
end

Dislin.disfin()

File: .vendor/dislin/go/src/areaf.go
package main

import "dislin"

func main  () {
  ix  := [4]int32{0, 300, 300,   0}
  iy  := [4]int32{0,   0, 400, 400}
  ixp := [4]int32{0, 0, 0, 0}
  iyp := [4]int32{0, 0, 0, 0}

  dislin.Metafl ("cons")
  dislin.Disini ()
  dislin.Setvlt ("small")
  dislin.Pagera ()
  dislin.Complx ()

  dislin.Height (50)
  ctit := "Shading patterns  (AREAF)"
  nl := dislin.Nlmess (ctit)
  dislin.Messag (ctit,  (2970 - nl)/2, 200)

  nx0 := 335
  ny0 := 350

  iclr := 0
  for i := 0; i < 3; i++ {
    ny := ny0 + i * 600

    for j := 0; j < 6; j++ {
      nx := nx0 + j * 400
      ii := i * 6 + j
      dislin.Shdpat (ii)
      iclr = iclr + 1
      iclr = iclr % 8
      if iclr == 0 {
        iclr = 8
      }
      dislin.Setclr (iclr)
      for k := 0; k < 4; k++ {
        ixp[k] = ix[k] + int32 (nx)
        iyp[k] = iy[k] + int32 (ny)
      }

      dislin.Areaf (&ixp[0], &iyp[0], 4)
      nl = dislin.Nlnumb (float64 (ii), -1)
      nx = nx +  (300 - nl) / 2
      dislin.Color ("foreground")
      dislin.Number (float64 (ii), -1, nx, ny + 460) 
    }
  }

  dislin.Disfin ()
}